# Question 29

**Type:** Basic Query - Mutual Exclusivity

**Question:** "Are mutations in CDKN2A, CDK4, and RB1 mutually exclusive in glioblastoma patients?"

**Answer:**

I'll help you analyze whether mutations in CDKN2A, CDK4, and RB1 are mutually exclusive in glioblastoma patients. Let me query the data to find co-occurrence patterns of these genes.



Calling `clickhouse_run_select_query` with args:
```json
{
  "query": "\n-- First, let's identify glioblastoma samples and check for mutations in CDKN2A, CDK4, and RB1\nWITH glioblastoma_samples AS (\n  SELECT DISTINCT sample_unique_id\n  FROM clinical_data_derived\n  WHERE cancer_study_identifier = 'msk_chord_2024'\n    AND attribute_name = 'CANCER_TYPE_DETAILED'\n    AND attribute_value LIKE '%Glioblastoma%'\n),\ngene_mutations AS (\n  SELECT \n    g.sample_unique_id,\n    g.hugo_gene_symbol,\n    1 as has_mutation\n  FROM genomic_event_derived g\n  INNER JOIN glioblastoma_samples gb ON g.sample_unique_id = gb.sample_unique_id\n  WHERE g.cancer_study_identifier = 'msk_chord_2024'\n    AND g.variant_type = 'mutation'\n    AND g.hugo_gene_symbol IN ('CDKN2A', 'CDK4', 'RB1')\n)\nSELECT \n  sample_unique_id,\n  sumIf(has_mutation, hugo_gene_symbol = 'CDKN2A') as has_CDKN2A,\n  sumIf(has_mutation, hugo_gene_symbol = 'CDK4') as has_CDK4,\n  sumIf(has_mutation, hugo_gene_symbol = 'RB1') as has_RB1\nFROM gene_mutations\nGROUP BY sample_unique_id\n"
}
```



Result from `clickhouse_run_select_query`:
```json
{"rows":[]}
```


Let me check what cancer types are available and verify the glioblastoma designation:



Calling `clickhouse_run_select_query` with args:
```json
{
  "query": "\n-- Check available cancer types in the study\nSELECT DISTINCT attribute_value, COUNT(*) as sample_count\nFROM clinical_data_derived\nWHERE cancer_study_identifier = 'msk_chord_2024'\n  AND attribute_name = 'CANCER_TYPE_DETAILED'\n  AND (attribute_value LIKE '%Glio%' OR attribute_value LIKE '%Brain%')\nGROUP BY attribute_value\nORDER BY sample_count DESC\n"
}
```



Result from `clickhouse_run_select_query`:
```json
{"rows":[]}
```


Let me check the broader cancer type category:



Calling `clickhouse_run_select_query` with args:
```json
{
  "query": "\n-- Check CANCER_TYPE for glioblastoma\nSELECT DISTINCT attribute_value, COUNT(*) as sample_count\nFROM clinical_data_derived\nWHERE cancer_study_identifier = 'msk_chord_2024'\n  AND attribute_name = 'CANCER_TYPE'\n  AND (attribute_value LIKE '%Glio%' OR attribute_value LIKE '%Brain%')\nGROUP BY attribute_value\nORDER BY sample_count DESC\n"
}
```



Result from `clickhouse_run_select_query`:
```json
{"rows":[]}
```


Let me check what cancer types are actually available in this study:



Calling `clickhouse_run_select_query` with args:
```json
{
  "query": "\n-- Check all cancer types in the study\nSELECT DISTINCT attribute_value, COUNT(*) as sample_count\nFROM clinical_data_derived\nWHERE cancer_study_identifier = 'msk_chord_2024'\n  AND attribute_name = 'CANCER_TYPE'\nGROUP BY attribute_value\nORDER BY sample_count DESC\nLIMIT 20\n"
}
```



Result from `clickhouse_run_select_query`:
```json
{"rows":[{"attribute_value":"Non-Small Cell Lung Cancer","sample_count":7809},{"attribute_value":"Colorectal Cancer","sample_count":5543},{"attribute_value":"Breast Cancer","sample_count":5368},{"attribute_value":"Prostate Cancer","sample_count":3211},{"attribute_value":"Pancreatic Cancer","sample_count":3109}]}
```


I see that the msk_chord_2024 study doesn't appear to contain glioblastoma samples. Let me search for studies that do contain glioblastoma data:



Calling `list_studies` with args:
```json
{
  "search": "glioblastoma",
  "limit": 20
}
```



Result from `list_studies`:
```json

```




Calling `clickhouse_run_select_query` with args:
```json
{
  "query": "\n-- Search for studies with glioblastoma in their name or description\nSELECT \n  cancer_study_identifier,\n  name,\n  description\nFROM cancer_study\nWHERE name LIKE '%Glio%' OR description LIKE '%Glio%' OR name LIKE '%GBM%'\nLIMIT 20\n"
}
```



Result from `clickhouse_run_select_query`:
```json
{"rows":[{"cancer_study_identifier":"difg_msk_2023","name":"IDH-mutated Diffuse Glioma (MSK, Clin Cancer Res 2024)","description":"Targeted sequencing of 73 IDH-mutated low-grade gliomas and their matched normal pairs via MSK-IMPACT. The data is available on <a href=\"https://www.synapse.org/#!Synapse:syn52658621/wiki/624065\">Synapse</a>."},{"cancer_study_identifier":"difg_glass_2019","name":"Diffuse Glioma (GLASS Consortium, Nature 2019)","description":"Whole genome or whole exome sequencing analysis of temporally separated diffuse glioma tumor/normal pairs from 222 adult patients."},{"cancer_study_identifier":"difg_glass","name":"Diffuse Glioma (GLASS Consortium)","description":"Whole genome or whole exome sequencing analysis of temporally separated diffuse glioma tumor/normal pairs from 329 adult patients. The data provided is based on the May 31, 2022 version and can be accessed at <a href=\"https://www.synapse.org/#!Synapse:syn17038081/wiki/585622\">Synapse</a>."},{"cancer_study_identifier":"glioma_mskcc_2019","name":"Glioma (MSK, Clin Cancer Res 2019)","description":"Targeted sequencing on MSK-IMPACT and FMI Panels of 1004 samples (837 with matched normals) from 923 glioma patients."},{"cancer_study_identifier":"glioma_msk_2018","name":"Glioma (MSK, Nature 2019)","description":"Targeted Sequencing of CSF samples (91 samples with 86 matched normals) from 42 glioma patients."},{"cancer_study_identifier":"lgg_tcga","name":"Brain Lower Grade Glioma (TCGA, Firehose Legacy)","description":"TCGA Brain Lower Grade Glioma. Source data from <A HREF=\"http://gdac.broadinstitute.org/runs/stddata__2016_01_28/data/LGG/20160128/\">GDAC Firehose</A>. Previously known as TCGA Provisional."},{"cancer_study_identifier":"lgg_ucsf_2014","name":"Low-Grade Gliomas (UCSF, Science 2014)","description":"Whole exome sequencing of 23 grade II glioma tumor/normal pairs."},{"cancer_study_identifier":"lgggbm_tcga_pub","name":"Merged Cohort of LGG and GBM (TCGA, Cell 2016)","description":"Whole exome sequencing of 1,122 LGG and GBM tumor/normal pairs."},{"cancer_study_identifier":"utuc_igbmc_2021","name":"Upper Tract Urothelial Carcinoma (IGBMC, Genome Biology 2021)","description":"Whole-exome sequencing of 30 upper-tract urothelial carcinoma patients and their matched normals."},{"cancer_study_identifier":"lgg_tcga_pan_can_atlas_2018","name":"Brain Lower Grade Glioma (TCGA, PanCancer Atlas)","description":"Brain Lower Grade Glioma TCGA PanCancer data. The original data is <a href=\"https://gdc.cancer.gov/about-data/publications/pancanatlas\">here</a>. The publications are <a href=\"https://www.cell.com/pb-assets/consortium/pancanceratlas/pancani3/index.html\">here</a>."},{"cancer_study_identifier":"difg_tcga_gdc","name":"Diffuse Glioma (TCGA, GDC)","description":"TCGA Diffuse Glioma. Source data from GDC and generated in Jul 2025 using Cancer Data Aggregator."},{"cancer_study_identifier":"gbm_tcga","name":"Glioblastoma Multiforme (TCGA, Firehose Legacy)","description":"TCGA Glioblastoma Multiforme. Source data from <A HREF=\"http://gdac.broadinstitute.org/runs/stddata__2016_01_28/data/GBM/20160128/\">GDAC Firehose</A>. Previously known as TCGA Provisional."},{"cancer_study_identifier":"gbm_tcga_pub","name":"Glioblastoma (TCGA, Nature 2008)","description":"Targeted sequencing in 91 of the 206 primary glioblastoma tumors (143 with matched normals) from the Cancer Genome Atlas (TCGA) Glioblastoma Project."},{"cancer_study_identifier":"gbm_tcga_pub2013","name":"Glioblastoma (TCGA, Cell 2013)","description":"Whole-exome and/or whole-genome sequencing of 291 of the 577 glioblastoma tumor/normal pairs. The Cancer Genome Atlas (TCGA) Glioblastoma Project."},{"cancer_study_identifier":"gbm_tcga_pan_can_atlas_2018","name":"Glioblastoma Multiforme (TCGA, PanCancer Atlas)","description":"Glioblastoma Multiforme TCGA PanCancer data. The original data is <a href=\"https://gdc.cancer.gov/about-data/publications/pancanatlas\">here</a>. The publications are <a href=\"https://www.cell.com/pb-assets/consortium/pancanceratlas/pancani3/index.html\">here</a>."},{"cancer_study_identifier":"gbm_columbia_2019","name":"Glioblastoma (Columbia, Nat Med. 2019)","description":"Whole-exome sequencing of 32 out of 42 glioblastomas patients with matched normals."},{"cancer_study_identifier":"gbm_cptac_2021","name":"Glioblastoma (CPTAC, Cell 2021)","description":"Proteogenomic and metabolomic characterization of human glioblastoma. Whole genome or whole exome sequencing of 99 samples. Generated by CPTAC."},{"cancer_study_identifier":"gbm_tcga_gdc","name":"Glioblastoma Multiforme (TCGA GDC, 2025)","description":"TCGA Glioblastoma Multiforme. Source data from <A HREF=\"https://gdc.cancer.gov\">NCI GDC</A> and generated in Aug 2025 using <A HREF=\"https://cda.readthedocs.io/en/latest/\">Cancer Data Aggregator</A>."}]}
```


Great! I found several glioblastoma studies. Let me analyze the TCGA PanCancer Atlas glioblastoma study (gbm_tcga_pan_can_atlas_2018) which is a comprehensive dataset:



Calling `clickhouse_run_select_query` with args:
```json
{
  "query": "\n-- Analyze mutual exclusivity of CDKN2A, CDK4, and RB1 mutations in glioblastoma\nWITH gene_mutations AS (\n  SELECT \n    sample_unique_id,\n    hugo_gene_symbol,\n    1 as has_mutation\n  FROM genomic_event_derived\n  WHERE cancer_study_identifier = 'gbm_tcga_pan_can_atlas_2018'\n    AND variant_type = 'mutation'\n    AND hugo_gene_symbol IN ('CDKN2A', 'CDK4', 'RB1')\n),\nsample_mutation_profile AS (\n  SELECT \n    sample_unique_id,\n    sumIf(has_mutation, hugo_gene_symbol = 'CDKN2A') > 0 as has_CDKN2A,\n    sumIf(has_mutation, hugo_gene_symbol = 'CDK4') > 0 as has_CDK4,\n    sumIf(has_mutation, hugo_gene_symbol = 'RB1') > 0 as has_RB1\n  FROM gene_mutations\n  GROUP BY sample_unique_id\n),\nall_samples AS (\n  SELECT DISTINCT sample_unique_id\n  FROM genomic_event_derived\n  WHERE cancer_study_identifier = 'gbm_tcga_pan_can_atlas_2018'\n),\ncomplete_profile AS (\n  SELECT \n    a.sample_unique_id,\n    COALESCE(s.has_CDKN2A, false) as has_CDKN2A,\n    COALESCE(s.has_CDK4, false) as has_CDK4,\n    COALESCE(s.has_RB1, false) as has_RB1\n  FROM all_samples a\n  LEFT JOIN sample_mutation_profile s ON a.sample_unique_id = s.sample_unique_id\n)\nSELECT \n  has_CDKN2A,\n  has_CDK4,\n  has_RB1,\n  COUNT(*) as sample_count\nFROM complete_profile\nGROUP BY has_CDKN2A, has_CDK4, has_RB1\nORDER BY sample_count DESC\n"
}
```



Result from `clickhouse_run_select_query`:
```json
{"rows":[{"has_CDKN2A":0,"has_CDK4":0,"has_RB1":0,"sample_count":547},{"has_CDKN2A":0,"has_CDK4":0,"has_RB1":1,"sample_count":37},{"has_CDKN2A":1,"has_CDK4":0,"has_RB1":0,"sample_count":3},{"has_CDKN2A":1,"has_CDK4":0,"has_RB1":1,"sample_count":1}]}
```


Now let me get summary statistics and check for statistical significance:



Calling `clickhouse_run_select_query` with args:
```json
{
  "query": "\n-- Summary statistics for each gene\nWITH gene_mutations AS (\n  SELECT \n    sample_unique_id,\n    hugo_gene_symbol,\n    1 as has_mutation\n  FROM genomic_event_derived\n  WHERE cancer_study_identifier = 'gbm_tcga_pan_can_atlas_2018'\n    AND variant_type = 'mutation'\n    AND hugo_gene_symbol IN ('CDKN2A', 'CDK4', 'RB1')\n),\nall_samples AS (\n  SELECT DISTINCT sample_unique_id\n  FROM genomic_event_derived\n  WHERE cancer_study_identifier = 'gbm_tcga_pan_can_atlas_2018'\n),\ntotal_count AS (\n  SELECT COUNT(*) as total_samples FROM all_samples\n)\nSELECT \n  hugo_gene_symbol,\n  COUNT(DISTINCT sample_unique_id) as mutated_samples,\n  total_samples,\n  ROUND(COUNT(DISTINCT sample_unique_id) * 100.0 / total_samples, 2) as mutation_percentage\nFROM gene_mutations, total_count\nGROUP BY hugo_gene_symbol, total_samples\nORDER BY mutated_samples DESC\n"
}
```



Result from `clickhouse_run_select_query`:
```json
{"rows":[{"hugo_gene_symbol":"RB1","mutated_samples":38,"total_samples":588,"mutation_percentage":6.46},{"hugo_gene_symbol":"CDKN2A","mutated_samples":4,"total_samples":588,"mutation_percentage":0.68}]}
```


Let me also check for CDK4 mutations specifically and look at co-occurrence patterns:



Calling `clickhouse_run_select_query` with args:
```json
{
  "query": "\n-- Detailed co-occurrence analysis\nWITH gene_mutations AS (\n  SELECT \n    sample_unique_id,\n    hugo_gene_symbol\n  FROM genomic_event_derived\n  WHERE cancer_study_identifier = 'gbm_tcga_pan_can_atlas_2018'\n    AND variant_type = 'mutation'\n    AND hugo_gene_symbol IN ('CDKN2A', 'CDK4', 'RB1')\n),\nall_samples AS (\n  SELECT DISTINCT sample_unique_id\n  FROM genomic_event_derived\n  WHERE cancer_study_identifier = 'gbm_tcga_pan_can_atlas_2018'\n)\nSELECT \n  'Total samples' as metric,\n  COUNT(*) as count\nFROM all_samples\nUNION ALL\nSELECT \n  'CDKN2A mutated' as metric,\n  COUNT(DISTINCT sample_unique_id) as count\nFROM gene_mutations\nWHERE hugo_gene_symbol = 'CDKN2A'\nUNION ALL\nSELECT \n  'CDK4 mutated' as metric,\n  COUNT(DISTINCT sample_unique_id) as count\nFROM gene_mutations\nWHERE hugo_gene_symbol = 'CDK4'\nUNION ALL\nSELECT \n  'RB1 mutated' as metric,\n  COUNT(DISTINCT sample_unique_id) as count\nFROM gene_mutations\nWHERE hugo_gene_symbol = 'RB1'\nUNION ALL\nSELECT \n  'CDKN2A + RB1 co-mutated' as metric,\n  COUNT(DISTINCT a.sample_unique_id) as count\nFROM gene_mutations a\nINNER JOIN gene_mutations b ON a.sample_unique_id = b.sample_unique_id\nWHERE a.hugo_gene_symbol = 'CDKN2A' AND b.hugo_gene_symbol = 'RB1'\nUNION ALL\nSELECT \n  'CDKN2A + CDK4 co-mutated' as metric,\n  COUNT(DISTINCT a.sample_unique_id) as count\nFROM gene_mutations a\nINNER JOIN gene_mutations b ON a.sample_unique_id = b.sample_unique_id\nWHERE a.hugo_gene_symbol = 'CDKN2A' AND b.hugo_gene_symbol = 'CDK4'\nUNION ALL\nSELECT \n  'CDK4 + RB1 co-mutated' as metric,\n  COUNT(DISTINCT a.sample_unique_id) as count\nFROM gene_mutations a\nINNER JOIN gene_mutations b ON a.sample_unique_id = b.sample_unique_id\nWHERE a.hugo_gene_symbol = 'CDK4' AND b.hugo_gene_symbol = 'RB1'\nUNION ALL\nSELECT \n  'All three co-mutated' as metric,\n  COUNT(DISTINCT a.sample_unique_id) as count\nFROM gene_mutations a\nINNER JOIN gene_mutations b ON a.sample_unique_id = b.sample_unique_id\nINNER JOIN gene_mutations c ON a.sample_unique_id = c.sample_unique_id\nWHERE a.hugo_gene_symbol = 'CDKN2A' \n  AND b.hugo_gene_symbol = 'CDK4' \n  AND c.hugo_gene_symbol = 'RB1'\n"
}
```



Result from `clickhouse_run_select_query`:
```json
{"rows":[{"metric":"Total samples","count":588},{"metric":"CDKN2A mutated","count":4},{"metric":"CDK4 mutated","count":0},{"metric":"RB1 mutated","count":38},{"metric":"CDKN2A + RB1 co-mutated","count":1},{"metric":"CDK4 + RB1 co-mutated","count":0},{"metric":"CDKN2A + CDK4 co-mutated","count":0},{"metric":"All three co-mutated","count":0}]}
```


Perfect! Now let me check if there are any copy number alterations (CNAs) in these genes, as they are often deleted rather than mutated:



Calling `clickhouse_run_select_query` with args:
```json
{
  "query": "\n-- Check for copy number alterations in CDKN2A, CDK4, and RB1\nSELECT \n  hugo_gene_symbol,\n  alteration,\n  COUNT(DISTINCT sample_id) as sample_count\nFROM cna c\nINNER JOIN sample s ON c.sample_id = s.internal_id\nINNER JOIN patient p ON s.patient_id = p.internal_id\nINNER JOIN cancer_study cs ON p.cancer_study_id = cs.cancer_study_id\nWHERE cs.cancer_study_identifier = 'gbm_tcga_pan_can_atlas_2018'\n  AND hugo_gene_symbol IN ('CDKN2A', 'CDK4', 'RB1')\n  AND alteration != 0\nGROUP BY hugo_gene_symbol, alteration\nORDER BY hugo_gene_symbol, alteration\n"
}
```



Result from `clickhouse_run_select_query`:
```json
{"error_message":"Query execution failed: Received ClickHouse exception, code: 60, server response: Code: 60. DB::Exception: Unknown table expression identifier 'cna' in scope SELECT hugo_gene_symbol, alteration, COUNTDistinct(sample_id) AS sample_count FROM cna AS c INNER JOIN sample AS s ON c.sample_id = s.internal_id INNER JOIN patient AS p ON s.patient_id = p.internal_id INNER JOIN cancer_study AS cs ON p.cancer_study_id = cs.cancer_study_id WHERE (cs.cancer_study_identifier = 'gbm_tcga_pan_can_atlas_2018') AND (hugo_gene_symbol IN ('CDKN2A', 'CDK4', 'RB1')) AND (alteration != 0) GROUP BY hugo_gene_symbol, alteration ORDER BY hugo_gene_symbol ASC, alteration ASC. (UNKNOWN_TABLE) (version 25.8.1.36912 (official build)) (for url https://dl96orhu96.us-east-1.aws.clickhouse.cloud:8443)"}
```




Calling `clickhouse_list_tables` with args:
```json
{}
```



Result from `clickhouse_list_tables`:
```json
{"tables":[{"name":"allele_specific_copy_number","total_rows":0,"comment":"Allele-specific CNA data for mutations. References mutation_event, genetic_profile, and sample."},{"name":"alteration_driver_annotation","total_rows":37356,"comment":"Annotations for genetic alterations classified as potential drivers. References genetic_profile and sample."},{"name":"cancer_study","total_rows":511,"comment":"Contains metadata for each cancer study in the portal. References type_of_cancer and reference_genome."},{"name":"cancer_study_tags","total_rows":0,"comment":"Tags associated with a cancer study. References cancer_study."},{"name":"clinical_attribute_meta","total_rows":18230,"comment":"Metadata describing clinical attributes. References cancer_study."},{"name":"clinical_data_derived","primary_key":"cancer_study_identifier, type, attribute_name, sample_unique_id","total_rows":11763690,"comment":"Unified clinical data table combining both patient-level and sample-level clinical attributes in a normalized key-value format. Uses FULL OUTER JOIN to ensure all defined clinical attributes appear for each patient/sample, even when no value is recorded (populated with empty string). Enables efficient querying of clinical data across different attribute types and levels."},{"name":"clinical_event","total_rows":2643379,"comment":"Time-bound clinical events for patients. References patient."},{"name":"clinical_event_data","total_rows":14395558,"comment":"Key-value attributes for clinical events. References clinical_event."},{"name":"clinical_event_derived","primary_key":"cancer_study_identifier, event_type, patient_unique_id","total_rows":14422476,"comment":"Denormalized clinical timeline events table combining temporal clinical events with their associated key-value attributes. Each row represents a specific attribute of a clinical event (e.g., treatment, surgery, specimen collection) with start/stop dates. Flattens the normalized clinical_event and clinical_event_data relationship into a queryable format for timeline analysis."},{"name":"clinical_patient","total_rows":4157292,"comment":"Patient-level clinical attribute values. References patient."},{"name":"clinical_sample","total_rows":5323875,"comment":"Sample-level clinical attribute values. References sample."},{"name":"cna_event","total_rows":77018,"comment":"Copy number alteration event definition. References gene."},{"name":"copy_number_seg","total_rows":28149618,"comment":"Raw segmented copy number data. References cancer_study and sample."},{"name":"copy_number_seg_file","total_rows":292,"comment":"File metadata for segmented copy number input. References cancer_study."},{"name":"gene","total_rows":46303,"comment":"Gene metadata (Entrez ID, HUGO symbol). References genetic_entity."},{"name":"gene_alias","total_rows":75207,"comment":"Alternative symbols/aliases for genes. References gene."},{"name":"gene_panel","total_rows":58,"comment":"Defines a targeted gene panel used in sequencing."},{"name":"gene_panel_list","total_rows":21477,"comment":"Join table linking gene panels to genes. References gene_panel and gene."},{"name":"generic_assay_data_derived","primary_key":"profile_type, entity_stable_id, patient_unique_id, sample_unique_id","total_rows":394482832,"comment":"Denormalized generic assay data table that unpivots wide-format genetic_alteration values for generic assay profiles into a long-format structure. Handles diverse assay types like treatment response, phosphosite quantification, and other non-genomic measurements. Each row represents one entity-sample combination with its measured value. Supports both sample-level and patient-level data based on profile configuration."},{"name":"generic_entity_properties","total_rows":2490012,"comment":"Properties for generic entities. References genetic_entity."},{"name":"geneset","total_rows":34837,"comment":"Collection of genes grouped into functional sets. References genetic_entity."},{"name":"geneset_gene","total_rows":8050756,"comment":"Join table linking genesets to genes. References geneset and gene."},{"name":"geneset_hierarchy_leaf","total_rows":0,"comment":"Mapping of hierarchy nodes to specific gene sets. References geneset_hierarchy_node and geneset."},{"name":"geneset_hierarchy_node","total_rows":0,"comment":"Hierarchy node for organizing gene sets. May reference parent node within same table."},{"name":"genetic_alteration","primary_key":"genetic_profile_id, genetic_entity_id","total_rows":33488996,"comment":"Stores genetic alteration values (for many genetic alteration types. e.g. MRNA_EXPRESSION, PROTEIN_LEVEL, GENERIC_ASSAY,...). References genetic_profile and genetic_entity."},{"name":"genetic_alteration_derived","primary_key":"cancer_study_identifier, hugo_gene_symbol, profile_type, sample_unique_id","total_rows":10133300301,"comment":"Denormalized genetic alteration data table that unpivots the wide-format genetic_alteration values into a long-format structure. Each row represents one gene-sample-profile combination with its alteration value. Excludes mutation data (handled separately) and generic assays, focusing on continuous/discrete data like expression, methylation, and copy number. Filters out \"NA\" values to reduce storage and improve query performance."},{"name":"genetic_entity","total_rows":2508068,"comment":"Abstract entity representing a gene, gene set, or other genomic element."},{"name":"genetic_profile","total_rows":2405,"comment":"Defines a molecular profile (e.g. mutations, CNA, mRNA expression). References cancer_study."},{"name":"genetic_profile_link","total_rows":2,"comment":"Links between genetic profiles. References genetic_profile (source and target)."},{"name":"genetic_profile_samples","total_rows":1896,"comment":"Stores ordered sample lists for a genetic profile (for many genetic alteration types. e.g. MRNA_EXPRESSION, PROTEIN_LEVEL, GENERIC_ASSAY,...). References genetic_profile."},{"name":"genomic_event_derived","primary_key":"genetic_profile_stable_id, cancer_study_identifier, variant_type, entrez_gene_id, hugo_gene_symbol, sample_unique_id","total_rows":48903328,"comment":"Unified genomic events table combining mutations, copy number alterations (CNA), and structural variants into a single denormalized view. Each row represents a gene-level genomic event observed in a sample. Structural variants are split into separate rows for site1 and site2 genes (excluding duplicates). Includes off-panel detection to identify events in genes not covered by the associated gene panel."},{"name":"gistic","total_rows":3547,"comment":"GISTIC-identified copy number alteration regions. References cancer_study."},{"name":"gistic_to_gene","total_rows":204806,"comment":"Mapping of GISTIC regions to genes. References gistic and gene."},{"name":"info","total_rows":1},{"name":"mut_sig","total_rows":2850,"comment":"MutSig significance analysis results. References cancer_study and gene."},{"name":"mutation","primary_key":"genetic_profile_id, entrez_gene_id","total_rows":17774534,"comment":"Mutation observations in specific samples and profiles. References mutation_event, gene, genetic_profile, and sample."},{"name":"mutation_count_by_keyword","total_rows":12451447,"comment":"Stores keyword-based aggregated mutation counts. References genetic_profile and gene."},{"name":"mutation_event","total_rows":8850403,"comment":"Defines unique mutation events (by position, gene, allele change). References gene."},{"name":"patient","total_rows":320721,"comment":"Stores patient-level identifiers of patients enrolled in a cancer study. References cancer_study."},{"name":"reference_genome","total_rows":3,"comment":"Reference genome definitions (e.g. hg19, GRCh38)."},{"name":"reference_genome_gene","total_rows":84612,"comment":"Mapping of reference genome builds to genes. References reference_genome and gene."},{"name":"resource_definition","total_rows":43,"comment":"Definitions of external resources (study, patient, sample level). References cancer_study."},{"name":"resource_patient","total_rows":10471,"comment":"Links external resources to patients. References patient."},{"name":"resource_sample","total_rows":3711,"comment":"Links external resources to samples. References sample."},{"name":"resource_study","total_rows":0,"comment":"Links external resources to cancer studies. References cancer_study."},{"name":"sample","total_rows":336873,"comment":"Biological samples collected from patients. References patient."},{"name":"sample_cna_event","primary_key":"genetic_profile_id, cna_event_id, sample_id","total_rows":30879655,"comment":"Observed CNA events per sample and profile. References cna_event, sample, and genetic_profile."},{"name":"sample_derived","primary_key":"cancer_study_identifier, sample_unique_id","total_rows":336873,"comment":"Denormalized sample table with globally unique identifiers and enriched metadata. Combines sample, patient, and cancer_study data with computed flags for sequencing status and copy number segment availability. Optimized for summary and detailed projections with base64-encoded identifiers for URL-safe usage."},{"name":"sample_list","total_rows":2727,"comment":"Named collections of samples within a study. References cancer_study."},{"name":"sample_list_list","total_rows":1452133,"comment":"Join table linking sample lists to samples. References sample and sample_list."},{"name":"sample_profile","total_rows":1090903,"comment":"Links samples to genetic profiles, optionally via a gene panel. References sample, genetic_profile, and gene_panel."},{"name":"sample_to_gene_panel_derived","primary_key":"gene_panel_id, alteration_type, genetic_profile_id, sample_unique_id","total_rows":1090903,"comment":"Denormalized mapping table linking samples to their associated gene panels for each genetic alteration type. Derived from sample_profile, genetic_profile, gene_panel, sample, and cancer_study tables. Defaults to WES (Whole Exome Sequencing) when no specific gene panel is assigned."},{"name":"structural_variant","total_rows":148760,"comment":"Structural variant data (fusions, translocations). References genetic_profile, sample, and gene (site1/site2)."},{"name":"type_of_cancer","total_rows":898,"comment":"Lookup table for cancer types."}]}
```


Let me check copy number alterations using the genomic_event_derived table:



Calling `clickhouse_run_select_query` with args:
```json
{
  "query": "\n-- Check for copy number alterations in CDKN2A, CDK4, and RB1 in glioblastoma\nWITH cna_events AS (\n  SELECT \n    sample_unique_id,\n    hugo_gene_symbol,\n    mutation_type as alteration_type\n  FROM genomic_event_derived\n  WHERE cancer_study_identifier = 'gbm_tcga_pan_can_atlas_2018'\n    AND variant_type = 'cna'\n    AND hugo_gene_symbol IN ('CDKN2A', 'CDK4', 'RB1')\n),\nall_samples AS (\n  SELECT DISTINCT sample_unique_id\n  FROM genomic_event_derived\n  WHERE cancer_study_identifier = 'gbm_tcga_pan_can_atlas_2018'\n)\nSELECT \n  hugo_gene_symbol,\n  alteration_type,\n  COUNT(DISTINCT sample_unique_id) as sample_count,\n  ROUND(COUNT(DISTINCT sample_unique_id) * 100.0 / (SELECT COUNT(*) FROM all_samples), 2) as percentage\nFROM cna_events\nGROUP BY hugo_gene_symbol, alteration_type\nORDER BY hugo_gene_symbol, sample_count DESC\n"
}
```



Result from `clickhouse_run_select_query`:
```json
{"rows":[{"hugo_gene_symbol":"CDK4","alteration_type":"NA","sample_count":82,"percentage":13.95},{"hugo_gene_symbol":"CDKN2A","alteration_type":"NA","sample_count":322,"percentage":54.76},{"hugo_gene_symbol":"RB1","alteration_type":"NA","sample_count":15,"percentage":2.55}]}
```


Now let me analyze the co-occurrence of both mutations AND copy number alterations:



Calling `clickhouse_run_select_query` with args:
```json
{
  "query": "\n-- Combined analysis of mutations and CNAs for CDKN2A, CDK4, and RB1\nWITH all_alterations AS (\n  SELECT \n    sample_unique_id,\n    hugo_gene_symbol,\n    variant_type\n  FROM genomic_event_derived\n  WHERE cancer_study_identifier = 'gbm_tcga_pan_can_atlas_2018'\n    AND hugo_gene_symbol IN ('CDKN2A', 'CDK4', 'RB1')\n    AND (\n      (variant_type = 'mutation') OR \n      (variant_type = 'cna' AND mutation_type = 'NA')  -- Deep deletions/amplifications\n    )\n),\nall_samples AS (\n  SELECT DISTINCT sample_unique_id\n  FROM genomic_event_derived\n  WHERE cancer_study_identifier = 'gbm_tcga_pan_can_atlas_2018'\n),\nsample_alteration_profile AS (\n  SELECT \n    sample_unique_id,\n    sumIf(1, hugo_gene_symbol = 'CDKN2A') > 0 as has_CDKN2A_alt,\n    sumIf(1, hugo_gene_symbol = 'CDK4') > 0 as has_CDK4_alt,\n    sumIf(1, hugo_gene_symbol = 'RB1') > 0 as has_RB1_alt\n  FROM all_alterations\n  GROUP BY sample_unique_id\n),\ncomplete_profile AS (\n  SELECT \n    a.sample_unique_id,\n    COALESCE(s.has_CDKN2A_alt, false) as has_CDKN2A_alt,\n    COALESCE(s.has_CDK4_alt, false) as has_CDK4_alt,\n    COALESCE(s.has_RB1_alt, false) as has_RB1_alt\n  FROM all_samples a\n  LEFT JOIN sample_alteration_profile s ON a.sample_unique_id = s.sample_unique_id\n)\nSELECT \n  has_CDKN2A_alt as CDKN2A_altered,\n  has_CDK4_alt as CDK4_altered,\n  has_RB1_alt as RB1_altered,\n  COUNT(*) as sample_count,\n  ROUND(COUNT(*) * 100.0 / (SELECT COUNT(*) FROM all_samples), 2) as percentage\nFROM complete_profile\nGROUP BY has_CDKN2A_alt, has_CDK4_alt, has_RB1_alt\nORDER BY sample_count DESC\n"
}
```



Result from `clickhouse_run_select_query`:
```json
{"rows":[{"CDKN2A_altered":1,"CDK4_altered":0,"RB1_altered":0,"sample_count":301,"percentage":51.19},{"CDKN2A_altered":0,"CDK4_altered":0,"RB1_altered":0,"sample_count":153,"percentage":26.02},{"CDKN2A_altered":0,"CDK4_altered":1,"RB1_altered":0,"sample_count":68,"percentage":11.56},{"CDKN2A_altered":0,"CDK4_altered":0,"RB1_altered":1,"sample_count":41,"percentage":6.97},{"CDKN2A_altered":1,"CDK4_altered":1,"RB1_altered":0,"sample_count":14,"percentage":2.38},{"CDKN2A_altered":1,"CDK4_altered":0,"RB1_altered":1,"sample_count":11,"percentage":1.87}]}
```


Excellent! Now let me calculate the statistical significance of mutual exclusivity:

---

## Model Information
- **model**: us.anthropic.claude-sonnet-4-5-20250929-v1:0
- **temperature**: 0.0
- **max_output_tokens**: 1024
- **thinking_enabled**: False
- **thinking_budget_tokens**: 1024
- **response_time_seconds**: 70.94227273296565
- **agent_type**: mcp-clickhouse

### Usage
- **input_tokens**: 101385
- **output_tokens**: 5622
- **total_tokens**: 107007
- **cache_creation_input_tokens**: 0
- **cache_read_input_tokens**: 0

---

*Generated on 2026-02-05 12:55:43*