# Question 52

**Type:** Genomic Alteration Frequencies

**Question:** What is the most frequent mutation in the TP53 gene in the TCGA breast cancer study?

**Answer:**

I'll help you find the most frequent mutation in the TP53 gene in the TCGA breast cancer study. Let me start by finding the relevant TCGA breast cancer study.



Calling `list_studies` with args:
```json
{
  "search": "TCGA breast"
}
```



Result from `list_studies`:
```json

```


Let me try a broader search:



Calling `list_studies` with args:
```json
{
  "search": "breast",
  "limit": 50
}
```



Result from `list_studies`:
```json
[{"cancer_study_identifier":"breast_msk_2025","name":"Breast Cancer (MSK, 2025)","type_of_cancer_id":"brca","sample_count":3880,"has_guide":false},{"cancer_study_identifier":"breast_msk_2018","name":"Breast Cancer (MSK, Cancer Cell 2018)","type_of_cancer_id":"breast","sample_count":1919,"has_guide":false},{"cancer_study_identifier":"breast_ink4_msk_2021","name":"Metastatic Breast Cancer (MSK, Cancer Discovery 2022)","type_of_cancer_id":"breast","sample_count":1366,"has_guide":false},{"cancer_study_identifier":"breast_cptac_gdc","name":"Breast Cancer (CPTAC GDC, 2025)","type_of_cancer_id":"breast","sample_count":155,"has_guide":false},{"cancer_study_identifier":"brca_mapk_hp_msk_2021","name":"MAPK on resistance to anti-HER2 therapy for breast cancer (MSK, Nat Commun. 2022)","type_of_cancer_id":"brca","sample_count":146,"has_guide":false},{"cancer_study_identifier":"breast_alpelisib_2020","name":"Breast Cancer (MSK, Nature Cancer 2020)","type_of_cancer_id":"breast","sample_count":142,"has_guide":false},{"cancer_study_identifier":"brca_cptac_2020","name":"Proteogenomic landscape of breast cancer (CPTAC, Cell 2020)","type_of_cancer_id":"brca","sample_count":123,"has_guide":false},{"cancer_study_identifier":"ilc_msk_2023","name":"Non-CDH1 Invasive Lobular Carcinoma (MSK, 2023)","type_of_cancer_id":"breast","sample_count":26,"has_guide":false}]
```


Let me search specifically for TCGA:



Calling `list_studies` with args:
```json
{
  "search": "TCGA",
  "limit": 50
}
```



Result from `list_studies`:
```json
[{"cancer_study_identifier":"pancan_pcawg_2020","name":"Pan-cancer analysis of whole genomes (ICGC/TCGA, Nature 2020)","type_of_cancer_id":"mixed","sample_count":2998,"has_guide":false},{"cancer_study_identifier":"nsclc_tcga_broad_2016","name":"Pan-Lung Cancer (TCGA, Nat Genet 2016)","type_of_cancer_id":"nsclc","sample_count":1145,"has_guide":false},{"cancer_study_identifier":"lgggbm_tcga_pub","name":"Merged Cohort of LGG and GBM (TCGA, Cell 2016)","type_of_cancer_id":"difg","sample_count":1123,"has_guide":false},{"cancer_study_identifier":"brca_tcga","name":"Breast Invasive Carcinoma (TCGA, Firehose Legacy)","type_of_cancer_id":"brca","sample_count":1109,"has_guide":false},{"cancer_study_identifier":"brca_tcga_gdc","name":"Invasive Breast Carcinoma (TCGA, GDC)","type_of_cancer_id":"brca","sample_count":1106,"has_guide":false},{"cancer_study_identifier":"brca_tcga_pan_can_atlas_2018","name":"Breast Invasive Carcinoma (TCGA, PanCancer Atlas)","type_of_cancer_id":"brca","sample_count":1085,"has_guide":true},{"cancer_study_identifier":"brca_tcga_pub","name":"Breast Invasive Carcinoma (TCGA, Nature 2012)","type_of_cancer_id":"brca","sample_count":826,"has_guide":false},{"cancer_study_identifier":"brca_tcga_pub2015","name":"Breast Invasive Carcinoma (TCGA, Cell 2015)","type_of_cancer_id":"brca","sample_count":819,"has_guide":false},{"cancer_study_identifier":"coadread_tcga","name":"Colorectal Adenocarcinoma (TCGA, Firehose Legacy)","type_of_cancer_id":"coadread","sample_count":641,"has_guide":false},{"cancer_study_identifier":"gbm_tcga","name":"Glioblastoma Multiforme (TCGA, Firehose Legacy)","type_of_cancer_id":"difg","sample_count":620,"has_guide":false},{"cancer_study_identifier":"ov_tcga","name":"Ovarian Serous Cystadenocarcinoma (TCGA, Firehose Legacy)","type_of_cancer_id":"hgsoc","sample_count":618,"has_guide":false},{"cancer_study_identifier":"hgsoc_tcga_gdc","name":"High-Grade Serous Ovarian Cancer (TCGA, GDC)","type_of_cancer_id":"hgsoc","sample_count":618,"has_guide":false},{"cancer_study_identifier":"gbm_tcga_gdc","name":"Glioblastoma Multiforme (TCGA GDC, 2025)","type_of_cancer_id":"difg","sample_count":612,"has_guide":false},{"cancer_study_identifier":"coadread_tcga_pan_can_atlas_2018","name":"Colorectal Adenocarcinoma (TCGA, PanCancer Atlas)","type_of_cancer_id":"coadread","sample_count":595,"has_guide":true},{"cancer_study_identifier":"gbm_tcga_pan_can_atlas_2018","name":"Glioblastoma Multiforme (TCGA, PanCancer Atlas)","type_of_cancer_id":"difg","sample_count":593,"has_guide":true},{"cancer_study_identifier":"luad_tcga_gdc","name":"Lung Adenocarcinoma (TCGA, GDC)","type_of_cancer_id":"luad","sample_count":588,"has_guide":false},{"cancer_study_identifier":"luad_tcga","name":"Lung Adenocarcinoma (TCGA, Firehose Legacy)","type_of_cancer_id":"luad","sample_count":587,"has_guide":false},{"cancer_study_identifier":"ov_tcga_pan_can_atlas_2018","name":"Ovarian Serous Cystadenocarcinoma (TCGA, PanCancer Atlas)","type_of_cancer_id":"hgsoc","sample_count":586,"has_guide":true},{"cancer_study_identifier":"gbm_tcga_pub2013","name":"Glioblastoma (TCGA, Cell 2013)","type_of_cancer_id":"difg","sample_count":578,"has_guide":false},{"cancer_study_identifier":"luad_tcga_pan_can_atlas_2018","name":"Lung Adenocarcinoma (TCGA, PanCancer Atlas)","type_of_cancer_id":"luad","sample_count":567,"has_guide":true},{"cancer_study_identifier":"stes_tcga_pub","name":"Esophageal Carcinoma (TCGA, Nature 2017)","type_of_cancer_id":"egc","sample_count":560,"has_guide":false},{"cancer_study_identifier":"ucec_tcga","name":"Uterine Corpus Endometrial Carcinoma (TCGA, Firehose Legacy)","type_of_cancer_id":"ucec","sample_count":550,"has_guide":false},{"cancer_study_identifier":"ucec_tcga_gdc","name":"Endometrial Carcinoma (TCGA, GDC)","type_of_cancer_id":"ucec","sample_count":550,"has_guide":false},{"cancer_study_identifier":"kirc_tcga","name":"Kidney Renal Clear Cell Carcinoma (TCGA, Firehose Legacy)","type_of_cancer_id":"ccrcc","sample_count":539,"has_guide":false},{"cancer_study_identifier":"ccrcc_tcga_gdc","name":"Renal Clear Cell Carcinoma (TCGA, GDC)","type_of_cancer_id":"ccrcc","sample_count":539,"has_guide":false},{"cancer_study_identifier":"difg_tcga_gdc","name":"Diffuse Glioma (TCGA, GDC)","type_of_cancer_id":"difg","sample_count":533,"has_guide":false},{"cancer_study_identifier":"hnsc_tcga","name":"Head and Neck Squamous Cell Carcinoma (TCGA, Firehose Legacy)","type_of_cancer_id":"hnsc","sample_count":531,"has_guide":false},{"cancer_study_identifier":"lgg_tcga","name":"Brain Lower Grade Glioma (TCGA, Firehose Legacy)","type_of_cancer_id":"difg","sample_count":531,"has_guide":false},{"cancer_study_identifier":"hnsc_tcga_gdc","name":"Head and Neck Squamous Cell Carcinoma (TCGA, GDC)","type_of_cancer_id":"hnsc","sample_count":531,"has_guide":false},{"cancer_study_identifier":"ucec_tcga_pan_can_atlas_2018","name":"Uterine Corpus Endometrial Carcinoma (TCGA, PanCancer Atlas)","type_of_cancer_id":"ucec","sample_count":530,"has_guide":true},{"cancer_study_identifier":"hnsc_tcga_pan_can_atlas_2018","name":"Head and Neck Squamous Cell Carcinoma (TCGA, PanCancer Atlas)","type_of_cancer_id":"hnsc","sample_count":524,"has_guide":false},{"cancer_study_identifier":"thca_tcga","name":"Thyroid Carcinoma (TCGA, Firehose Legacy)","type_of_cancer_id":"thpa","sample_count":517,"has_guide":false},{"cancer_study_identifier":"thpa_tcga_gdc","name":"Papillary Thyroid Cancer (TCGA, GDC)","type_of_cancer_id":"thpa","sample_count":516,"has_guide":false},{"cancer_study_identifier":"lgg_tcga_pan_can_atlas_2018","name":"Brain Lower Grade Glioma (TCGA, PanCancer Atlas)","type_of_cancer_id":"difg","sample_count":515,"has_guide":false},{"cancer_study_identifier":"kirc_tcga_pan_can_atlas_2018","name":"Kidney Renal Clear Cell Carcinoma (TCGA, PanCancer Atlas)","type_of_cancer_id":"ccrcc","sample_count":513,"has_guide":false},{"cancer_study_identifier":"lusc_tcga","name":"Lung Squamous Cell Carcinoma (TCGA, Firehose Legacy)","type_of_cancer_id":"lusc","sample_count":512,"has_guide":false},{"cancer_study_identifier":"lusc_tcga_gdc","name":"Lung Squamous Cell Carcinoma (TCGA, GDC)","type_of_cancer_id":"lusc","sample_count":505,"has_guide":false},{"cancer_study_identifier":"prad_tcga","name":"Prostate Adenocarcinoma (TCGA, Firehose Legacy)","type_of_cancer_id":"prad","sample_count":502,"has_guide":false},{"cancer_study_identifier":"prad_tcga_gdc","name":"Prostate Adenocarcinoma (TCGA, GDC)","type_of_cancer_id":"prad","sample_count":502,"has_guide":false},{"cancer_study_identifier":"thca_tcga_pan_can_atlas_2018","name":"Thyroid Carcinoma (TCGA, PanCancer Atlas)","type_of_cancer_id":"thpa","sample_count":501,"has_guide":false},{"cancer_study_identifier":"thca_tcga_pub","name":"Papillary Thyroid Carcinoma (TCGA, Cell 2014)","type_of_cancer_id":"thpa","sample_count":497,"has_guide":false},{"cancer_study_identifier":"prad_tcga_pan_can_atlas_2018","name":"Prostate Adenocarcinoma (TCGA, PanCancer Atlas)","type_of_cancer_id":"prad","sample_count":495,"has_guide":false},{"cancer_study_identifier":"ov_tcga_pub","name":"Ovarian Serous Cystadenocarcinoma (TCGA, Nature 2011)","type_of_cancer_id":"hgsoc","sample_count":490,"has_guide":false},{"cancer_study_identifier":"lusc_tcga_pan_can_atlas_2018","name":"Lung Squamous Cell Carcinoma (TCGA, PanCancer Atlas)","type_of_cancer_id":"lusc","sample_count":488,"has_guide":false},{"cancer_study_identifier":"skcm_tcga","name":"Skin Cutaneous Melanoma (TCGA, Firehose Legacy)","type_of_cancer_id":"skcm","sample_count":481,"has_guide":false},{"cancer_study_identifier":"stad_tcga","name":"Stomach Adenocarcinoma (TCGA, Firehose Legacy)","type_of_cancer_id":"stad","sample_count":479,"has_guide":false},{"cancer_study_identifier":"blca_msk_tcga_2020","name":"Bladder Cancer (MSK/TCGA, 2020)","type_of_cancer_id":"blca","sample_count":477,"has_guide":false},{"cancer_study_identifier":"skcm_tcga_gdc","name":"Cutaneous Melanoma (TCGA, GDC)","type_of_cancer_id":"skcm","sample_count":475,"has_guide":false},{"cancer_study_identifier":"coad_tcga_gdc","name":"Colon Adenocarcinoma (TCGA, GDC)","type_of_cancer_id":"coad","sample_count":465,"has_guide":false},{"cancer_study_identifier":"skcm_tcga_pan_can_atlas_2018","name":"Skin Cutaneous Melanoma (TCGA, PanCancer Atlas)","type_of_cancer_id":"skcm","sample_count":449,"has_guide":false}]
```


Great! I found several TCGA breast cancer studies. The most comprehensive one appears to be "brca_tcga_pan_can_atlas_2018" (Breast Invasive Carcinoma, TCGA PanCancer Atlas) with 1085 samples. Let me query for TP53 mutations in this study:



Calling `clickhouse_run_select_query` with args:
```json
{
  "query": "\nSELECT \n    protein_change,\n    COUNT(*) as mutation_count,\n    COUNT(DISTINCT sample_unique_id) as sample_count\nFROM genomic_event_derived\nWHERE cancer_study_identifier = 'brca_tcga_pan_can_atlas_2018'\n    AND hugo_gene_symbol = 'TP53'\n    AND variant_type = 'mutation'\n    AND protein_change != ''\nGROUP BY protein_change\nORDER BY mutation_count DESC\nLIMIT 20\n"
}
```



Result from `clickhouse_run_select_query`:
```json
{"error_message":"Query execution failed: Received ClickHouse exception, code: 47, server response: Code: 47. DB::Exception: Unknown expression identifier `protein_change` in scope SELECT protein_change, COUNT(*) AS mutation_count, COUNTDistinct(sample_unique_id) AS sample_count FROM genomic_event_derived WHERE (cancer_study_identifier = 'brca_tcga_pan_can_atlas_2018') AND (hugo_gene_symbol = 'TP53') AND (variant_type = 'mutation') AND (protein_change != '') GROUP BY protein_change ORDER BY mutation_count DESC LIMIT 20. (UNKNOWN_IDENTIFIER) (version 25.8.1.36912 (official build)) (for url https://dl96orhu96.us-east-1.aws.clickhouse.cloud:8443)"}
```


Let me check the available columns in the genomic_event_derived table:



Calling `clickhouse_list_table_columns` with args:
```json
{
  "table": "genomic_event_derived"
}
```



Result from `clickhouse_list_table_columns`:
```json
{"columns":[{"name":"sample_unique_id","type":"String","comment":"Globally unique sample identifier constructed as cancer_study_identifier + \"_\" + sample.stable_id to ensure uniqueness across all studies."},{"name":"hugo_gene_symbol","type":"String","comment":"Official HUGO gene symbol for the affected gene. References gene.hugo_gene_symbol."},{"name":"entrez_gene_id","type":"Int32","comment":"Entrez Gene ID for the affected gene. References gene.entrez_gene_id."},{"name":"gene_panel_stable_id","type":"LowCardinality(String)","comment":"Identifier of the gene panel used for this sample. Defaults to \"WES\" (Whole Exome Sequencing) when no specific panel is assigned."},{"name":"cancer_study_identifier","type":"LowCardinality(String)","comment":"Stable string identifier for the cancer study. References cancer_study.cancer_study_identifier."},{"name":"genetic_profile_stable_id","type":"LowCardinality(String)","comment":"Stable identifier of the genetic profile containing this event. References genetic_profile.stable_id."},{"name":"variant_type","type":"LowCardinality(String)","comment":"Type of genomic variant: \"mutation\" for point mutations/indels, \"cna\" for copy number alterations, \"structural_variant\" for structural rearrangements."},{"name":"mutation_variant","type":"String","comment":"Protein-level change for mutations (e.g., p.V600E). Set to \"NA\" for non-mutation events. References mutation_event.protein_change."},{"name":"mutation_type","type":"LowCardinality(String)","comment":"Classification of mutation type (e.g., Missense, Nonsense, Frame_Shift). Set to \"NA\" for non-mutation events. References mutation_event.mutation_type."},{"name":"mutation_status","type":"LowCardinality(String)","comment":"Mutation status: Germline, Somatic, or LOH. Set to \"NA\" for non-mutation events. References mutation.mutation_status."},{"name":"driver_filter","type":"LowCardinality(String)","comment":"Driver annotation filter applied to the alteration. Currently set to \"NA\" for all events (driver annotations not yet implemented)."},{"name":"driver_tiers_filter","type":"LowCardinality(String)","comment":"Driver tiers annotation filter. Currently set to \"NA\" for all events (driver annotations not yet implemented)."},{"name":"cna_alteration","type":"Nullable(Int8)","comment":"Copy number alteration value (e.g., -2 for deep deletion, +2 for amplification). NULL for non-CNA events. References cna_event.alteration."},{"name":"cna_cytoband","type":"String","comment":"Cytogenetic band location of the gene for CNA events. Empty string for non-CNA events. References reference_genome_gene.cytoband."},{"name":"sv_event_info","type":"String","comment":"Additional information about structural variant events. Empty string for non-SV events. References structural_variant.event_info."},{"name":"patient_unique_id","type":"String","comment":"Globally unique patient identifier constructed as cancer_study_identifier + \"_\" + patient.stable_id to ensure uniqueness across all studies."},{"name":"off_panel","type":"Bool","comment":"Boolean flag indicating whether this genomic event occurred in a gene not covered by the associated gene panel (TRUE=off-panel, FALSE=on-panel). Determined by checking presence in gene_panel_to_gene_derived table."}]}
```


Perfect! The column is called `mutation_variant`, not `protein_change`. Let me query again:



Calling `clickhouse_run_select_query` with args:
```json
{
  "query": "\nSELECT \n    mutation_variant,\n    mutation_type,\n    COUNT(*) as mutation_count,\n    COUNT(DISTINCT sample_unique_id) as sample_count\nFROM genomic_event_derived\nWHERE cancer_study_identifier = 'brca_tcga_pan_can_atlas_2018'\n    AND hugo_gene_symbol = 'TP53'\n    AND variant_type = 'mutation'\n    AND mutation_variant != 'NA'\n    AND mutation_variant != ''\nGROUP BY mutation_variant, mutation_type\nORDER BY mutation_count DESC\nLIMIT 20\n"
}
```



Result from `clickhouse_run_select_query`:
```json
{"rows":[{"mutation_variant":"R175H","mutation_type":"Missense_Mutation","mutation_count":21,"sample_count":21},{"mutation_variant":"R273H","mutation_type":"Missense_Mutation","mutation_count":13,"sample_count":13},{"mutation_variant":"R196*","mutation_type":"Nonsense_Mutation","mutation_count":8,"sample_count":8},{"mutation_variant":"R213*","mutation_type":"Nonsense_Mutation","mutation_count":7,"sample_count":7},{"mutation_variant":"R342*","mutation_type":"Nonsense_Mutation","mutation_count":7,"sample_count":7},{"mutation_variant":"Y220C","mutation_type":"Missense_Mutation","mutation_count":7,"sample_count":7},{"mutation_variant":"H193R","mutation_type":"Missense_Mutation","mutation_count":6,"sample_count":6},{"mutation_variant":"H179R","mutation_type":"Missense_Mutation","mutation_count":6,"sample_count":6},{"mutation_variant":"X225_splice","mutation_type":"Splice_Site","mutation_count":5,"sample_count":5},{"mutation_variant":"R248W","mutation_type":"Missense_Mutation","mutation_count":5,"sample_count":5},{"mutation_variant":"R273C","mutation_type":"Missense_Mutation","mutation_count":5,"sample_count":5},{"mutation_variant":"I195T","mutation_type":"Missense_Mutation","mutation_count":5,"sample_count":5},{"mutation_variant":"R209Kfs*6","mutation_type":"Frame_Shift_Del","mutation_count":4,"sample_count":4},{"mutation_variant":"C176F","mutation_type":"Missense_Mutation","mutation_count":4,"sample_count":4},{"mutation_variant":"X126_splice","mutation_type":"Splice_Site","mutation_count":4,"sample_count":4},{"mutation_variant":"R282W","mutation_type":"Missense_Mutation","mutation_count":4,"sample_count":4},{"mutation_variant":"L194R","mutation_type":"Missense_Mutation","mutation_count":4,"sample_count":4},{"mutation_variant":"E285K","mutation_type":"Missense_Mutation","mutation_count":4,"sample_count":4},{"mutation_variant":"R248Q","mutation_type":"Missense_Mutation","mutation_count":4,"sample_count":4},{"mutation_variant":"V216M","mutation_type":"Missense_Mutation","mutation_count":3,"sample_count":3}]}
```


Perfect! Now let me get some additional context about the total number of TP53 mutations and samples:



Calling `clickhouse_run_select_query` with args:
```json
{
  "query": "\nSELECT \n    COUNT(*) as total_mutations,\n    COUNT(DISTINCT sample_unique_id) as total_samples_with_tp53_mutation\nFROM genomic_event_derived\nWHERE cancer_study_identifier = 'brca_tcga_pan_can_atlas_2018'\n    AND hugo_gene_symbol = 'TP53'\n    AND variant_type = 'mutation'\n"
}
```



Result from `clickhouse_run_select_query`:
```json
{"rows":[{"total_mutations":353,"total_samples_with_tp53_mutation":347}]}
```


Excellent! Now let me also check the total number of samples in the study to calculate the mutation frequency:



Calling `clickhouse_run_select_query` with args:
```json
{
  "query": "\nSELECT COUNT(DISTINCT sample_unique_id) as total_samples\nFROM genomic_event_derived\nWHERE cancer_study_identifier = 'brca_tcga_pan_can_atlas_2018'\n"
}
```



Result from `clickhouse_run_select_query`:
```json
{"rows":[{"total_samples":1080}]}
```


Perfect! Now I have all the information needed to answer your question.

## Answer

The most frequent mutation in the TP53 gene in the TCGA breast cancer study (Breast Invasive Carcinoma, TCGA PanCancer Atlas) is **R175H**, a missense mutation that occurs in **21 samples**.

### Key Findings:

**Top 5 Most Frequent TP53 Mutations:**
1. **R175H** (Missense) - 21 samples
2. **R273H** (Missense) - 13 samples  
3. **R196*** (Nonsense) - 8 samples
4. **R213*** (Nonsense) - 7 samples
5. **R342*** (Nonsense) - 7 samples

### Context:
- **Total TP53 mutations**: 353 mutations across 347 samples
- **Total samples in study**: 1,080 samples
- **TP53 mutation frequency**: ~32% of samples (347/1,080)

The R175H mutation is a well-known hotspot mutation in TP53 that affects the DNA-binding domain of the p53 protein. This mutation is one of the most common TP53 alterations across multiple cancer types and is associated with loss of tumor suppressor function. The high frequency of R273H (another DNA-binding domain mutation) as the second most common mutation is also consistent with known TP53 mutation patterns in breast cancer.

---

## Model Information
- **model**: us.anthropic.claude-sonnet-4-5-20250929-v1:0
- **temperature**: 0.0
- **max_output_tokens**: 1024
- **thinking_enabled**: False
- **thinking_budget_tokens**: 1024
- **response_time_seconds**: 33.102795519982465
- **agent_type**: mcp-clickhouse

### Usage
- **input_tokens**: 63629
- **output_tokens**: 1463
- **total_tokens**: 65092
- **cache_creation_input_tokens**: 0
- **cache_read_input_tokens**: 0

---

*Generated on 2026-02-05 13:18:57*